### PR TITLE
fix:core: c4 audit m05

### PR DESCRIPTION
**Pull request by nmimran99**
fix: c4 audit m-05 ticket

Added a check to verify that `finalLongPayout` will not be set more than once.

### CODE REVIEW

File: apps/smart-contracts/core/test/unit/PrePOMarket.test.ts [hunk 1]
```diff
@@ -226,14 +226,13 @@ describe('=> prePOMarket', () => {
       expect(await prePOMarket.getFinalLongPayout()).to.eq(TEST_CEILING_PAYOUT.sub(1))
     })
 
-    it('should correctly set the same value twice', async () => {
+    it('reverts if setting the value more than once', async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/PrePOMarket.test.ts:229]: nit:
```suggestion
    it('reverts if value set more than once', async () => {
```

Comment (XavierEkkel) [reply apps/smart-contracts/core/test/unit/PrePOMarket.test.ts:229]: and move this test below the role revert test to keep the right order of reverts

**Review by XavierEkkel — APPROVED** [review]
LGTM after nits

**Review by ramenforbreakfast — APPROVED** [review]
LGTM

**Pull request merged**